File: aexample/src/main/java/org/judal/examples/java/model/StudentDocument.java
package org.judal.examples.java.model;

import javax.jdo.JDOException;

import org.judal.storage.EngineFactory;
import org.judal.storage.java.MapRecord;
import org.judal.storage.relational.RelationalDataSource;

public class StudentDocument extends MapRecord {

	private static final long serialVersionUID = 1L;
	
	public static final String TABLE_NAME = "student_document";
	
	public StudentDocument() throws JDOException {
		this(EngineFactory.getDefaultRelationalDataSource());
	}

	public StudentDocument(RelationalDataSource dataSource) throws JDOException {
		super(dataSource, TABLE_NAME);
	}

}

File: aexample/src/main/scala/org/judal/examples/scala/model/StudentDocument.scala
package org.judal.examples.scala.model

import javax.jdo.JDOException

import org.judal.storage.EngineFactory
import org.judal.storage.scala.MapRecord
import org.judal.storage.relational.RelationalDataSource

class StudentDocument(dataSource: RelationalDataSource) extends MapRecord(dataSource, StudentDocument.TABLE_NAME) {

	def this() = this(EngineFactory.getDefaultRelationalDataSource())

}

object StudentDocument {
  val TABLE_NAME = "student_document"
}